external/llvm-project/mlir/lib/Conversion/GPUToROCDL/LowerGpuOpsToROCDLOps.cpp

```diff
@@ -720,6 +720,31 @@ struct SoftwareBF16Trunc : OpRewritePattern<LLVM::FPTruncOp> {
     return success();
   }
 };
+
+struct LDSBarrierOpLowering : ConvertToLLVMPattern {
+  explicit LDSBarrierOpLowering(MLIRContext *context,
+                                LLVMTypeConverter &typeConverter)
+      : ConvertToLLVMPattern(gpu::LDSBarrierOp::getOperationName(), context,
```

Comment:
These days, there's a `ConvertOpToLLVMPattern` that provides better ergonomics and is preferred

Reply:
Adopted the pattern in the latest commit.